### PR TITLE
Changes to PS Deployment Script cmdlets

### PR DESCRIPTION
**Pull request by Xynoclafe**
## Description

* Formatted output property and show it on the Az-GetDeploymentScript cmdlet output
* Renamed -DeploymentScriptInputObject parameter name to -DeploymentScriptObject (DeploymentScriptInputObject  remains as an alias)
* Fixed missing target name in what if result - Performing the operation "Remove Deployment Script" on target "".
* Fix missing file name in message: "A file with the name '' already exists in directory 'C:\'. This operation will replace the existing file. Would you like to continue?"
* Ensure correct line endings are used in Az-SaveDeploymentScriptLog

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Resources/ResourceManager/Implementation/DeploymentScripts/SaveAzDeploymentScriptLog.cs [hunk 1]
```diff
@@ -118,11 +122,16 @@ public override void ExecuteCmdlet()
...
                         this.Force || !AzureSession.Instance.DataStore.FileExists(outputPathWithFileName),
                         string.Format(
-                            Properties.Resources.DeploymentScriptLogFileExists, Name, OutputPath),
+                            Properties.Resources.DeploymentScriptLogFileExists, scriptName, OutputPath),
                         Properties.Resources.DeploymentScriptShouldProcessString,
                         OutputPath,
                         () =>
                         {
+                            //Standardize newline character to be written into file
+                            string[] delims = new string[] { "\r", "\n", "\r\n" };
```
Comment (filizt) [inline src/Resources/ResourceManager/Implementation/DeploymentScripts/SaveAzDeploymentScriptLog.cs:131]: This is good but just curious did StringReader not work?

File: src/Resources/ResourceManager/SdkExtensions/ResourcesExtensions.cs [hunk 2]
```diff
@@ -195,6 +195,38 @@ private static string ConstructTemplateLinkView(TemplateLink templateLink)
...
+            dictionary.Values.ForEach(v => maxTypeLength = Math.Max(maxTypeLength, v.ToString().Split('\n').OrderByDescending(s => s.Length).First().Length + 2));
+
+            StringBuilder output = new StringBuilder();
+                
+                if (dictionary.Count > 0)
+                {
+                    string rowFormat = "{0, -" + maxNameLength + "}  {1, -" + maxTypeLength + "}\r\n";
+                    output.AppendLine();
+                    output.AppendFormat(rowFormat, "Key", "Value");
+                    output.AppendFormat(rowFormat, GeneralUtilities.GenerateSeparator(maxNameLength, "="), GeneralUtilities.GenerateSeparator(maxNameLength, "="));
```
Comment (filizt) [inline src/Resources/ResourceManager/SdkExtensions/ResourcesExtensions.cs:218]: > GeneralUtilities.GenerateSeparator(maxNameLength, "=") [](start = 106, length = 55)

A bit confused with the use of `maxTypeLength`. You don't seem to use it in the AppendFormat, do we need it then?

File: src/Resources/ResourceManager/Implementation/DeploymentScripts/SaveAzDeploymentScriptLog.cs (hunk 1 above)
Comment (Xynoclafe) [reply src/Resources/ResourceManager/Implementation/DeploymentScripts/SaveAzDeploymentScriptLog.cs:131]: I didn't really try StringReader. I wasn't sure if ReadLine() would consider all 3 possible line endings or only Environment.newline.

File: src/Resources/ResourceManager/SdkExtensions/ResourcesExtensions.cs (hunk 2 above)
Comment (Xynoclafe) [reply src/Resources/ResourceManager/SdkExtensions/ResourcesExtensions.cs:218]: Ah yes, I think I can get rid of that variable. I was using it earlier to determine the length of the separating line at the top. It's redundant now.

File: src/Resources/ResourceManager/Implementation/DeploymentScripts/SaveAzDeploymentScriptLog.cs (hunk 1 above)
Comment (filizt) [reply src/Resources/ResourceManager/Implementation/DeploymentScripts/SaveAzDeploymentScriptLog.cs:131]: It should consider all 3 and end of stream marker. See here: https://docs.microsoft.com/en-us/dotnet/api/system.io.stringreader.readline?view=netcore-3.1

Comment (Xynoclafe) [reply src/Resources/ResourceManager/Implementation/DeploymentScripts/SaveAzDeploymentScriptLog.cs:131]: Oh! let me quickly try that out as well

**Pull request merged**